index.js

```diff
@@ -449,6 +463,24 @@ class AbstractBtpPlugin extends EventEmitter {
   ilpAndCustomToProtocolData (obj) {
     return ilpAndCustomToProtocolData(obj)
   }
+
+  _emitDisconnect () {
+    if (this._readyState !== READY_STATES.disconnected) {
+      this._readyState = READY_STATES.disconnected
+      this._connected = false
+      this.emit('disconnect')
+    }
+  }
+
+  _emitConnect () {
+    if (this._readyState < READY_STATES.connected) {
```

Comment:
It's dangerous to rely on the ordering of enum values. Might be better to say explicitly `state !== connected && state !== readyToEmit`.

If you keep it this way, I would at least add a comment where the enum is defined to call out that you're relying on the ordering.